src/Entity/ListBuilder/AppListBuilder.php

```diff
@@ -316,6 +327,14 @@ protected function checkAppCredentialWarnings(AppInterface $app): array {
         }
         break;
       }
+
+      // Check for expired credentials.
+      if (($expired_date = $credential->getExpiresAt()) && \Drupal::time()->getRequestTime() - intval($expired_date->getTimestamp() / 1000) > 0) {
```

Comment:
Do not use inline variable assignment in conditions.
`($expired_date = $credential->getExpiresAt()) ` 
Also, the newly introduced value is just a shortcut for `$credential->getExpiresAt() `

Reply:
`$credential->getExpiresAt()` can be null.

Reply:
Yes, it can be `null` but it should not prevent you from moving this condition above where all values are calculated, above this line:

```php
    // Below, $value is expected to be a string in some places, and it might be
    // TranslatableMarkup or a string. If it is not a string, then in some
    // cases warnings will be generated. This way it is always a string,
    // removing ambiguity.
    $value = (string) $value;
```

The code block where it is now is responsible for the rendering of the (calculated) `$value`, this is my only problem with it.